ext/cdi/jersey-cdi1x/src/main/java/org/glassfish/jersey/ext/cdi1x/internal/CdiComponentProvider.java

```diff
@@ -674,9 +677,21 @@ public void inject(final Object t, final CreationalContext cc) {
             }
 
             delegate.inject(t, cc); // here the injection manager is used in HK2Bean
-
-            if (injectingManager != null) {
-                injectingManager.inject(t, CdiComponentProvider.CDI_CLASS_ANALYZER);
+            try {
+                if (injectingManager != null) {
+                    injectingManager.inject(t, CdiComponentProvider.CDI_CLASS_ANALYZER);
+                }
+            } catch (IllegalStateException e) {
+                if (JERSEY_CLASS_ANALYZER_REATTEMPT_INJECTION && e.getMessage().contains("ClassAnalyzer")) {
```

Comment:
Can you not just use the `CDI_CLASS_ANALYZER` name? 
Or does it not exactly match that name?

Reply:
@Pandrex247 yes I used that constant instead of my text to compara the content of the message, also I noticed that I can't use only the IllegalStateException, on some cases the reported exception was MultiException and others EJBException that is why I left the more generic case and I evaluate the content of the message with the constan value of the class analuzer name